### PR TITLE
Add .pick and .assign methods to Runnable

### PR DESCRIPTION
**Pull request by nfcampos**
Fixes # (issue)

### CODE REVIEW

File: langchain-core/src/runnables/base.ts [hunk 1]
```diff
@@ -540,6 +540,31 @@ export abstract class Runnable<
...
+  pick(keys: string | string[]): RunnableSequence {
+    // eslint-disable-next-line @typescript-eslint/no-use-before-define, @typescript-eslint/no-explicit-any
+    return this.pipe(new RunnablePick(keys) as any);
+  }
+
+  /**
+   * Assigns new fields to the dict output of this runnable. Returns a new runnable.
+   */
+  assign(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (jacoblee93) [inline langchain-core/src/runnables/base.ts:555]: Don't need this linter comment

File: langchain-core/src/runnables/passthrough.ts [hunk 2]
```diff
@@ -1,206 +1,12 @@
...
-  ): Promise<IterableReadableStream<RunOutput>> {
-    async function* generator() {
-      yield input;
-    }
-    return IterableReadableStream.fromAsyncGenerator(
-      this.transform(generator(), options)
-    );
-  }
-}
+export { RunnableAssign };
```
Comment (jacoblee93) [inline langchain-core/src/runnables/passthrough.ts:9]: I think you can remove this export? Only entrypoint is through `index.ts`

File: langchain-core/src/runnables/base.ts [hunk 3]
```diff
@@ -540,6 +540,31 @@ export abstract class Runnable<
     });
   }
 
+  /**
+   * Pick keys from the dict output of this runnable. Returns a new runnable.
+   */
+  pick(keys: string | string[]): RunnableSequence {
+    // eslint-disable-next-line @typescript-eslint/no-use-before-define, @typescript-eslint/no-explicit-any
+    return this.pipe(new RunnablePick(keys) as any);
```
Comment (jacoblee93) [inline langchain-core/src/runnables/base.ts:548]: Do we need the cast? And if we do, `as Runnable` is better?

File: langchain-core/src/runnables/passthrough.ts (hunk 2 above)
Comment (nfcampos) [reply langchain-core/src/runnables/passthrough.ts]: Done

File: langchain-core/src/runnables/base.ts (hunk 3 above)
Comment (nfcampos) [reply langchain-core/src/runnables/base.ts]: we need because of the dict input restriction

**Pull request merged**